security/kubernetes-nodes.md

```diff
@@ -0,0 +1,280 @@
+```
+
+If successful, host endpoints are created for each of your cluster's nodes:
+
+```bash
+calicoctl get heps -owide
+```
+
+The output may look similar to this:
+
+```
+$ calicoctl get heps -owide
+NAME                                                    NODE                                           INTERFACE   IPS                              PROFILES
+ip-172-16-101-147.us-west-2.compute.internal-auto-hep   ip-172-16-101-147.us-west-2.compute.internal   *           172.16.101.147,192.168.228.128   projectcalico-default-allow
+ip-172-16-101-54.us-west-2.compute.internal-auto-hep    ip-172-16-101-54.us-west-2.compute.internal    *           172.16.101.54,192.168.107.128    projectcalico-default-allow
+ip-172-16-101-79.us-west-2.compute.internal-auto-hep    ip-172-16-101-79.us-west-2.compute.internal    *           172.16.101.79,192.168.91.64      projectcalico-default-allow
+ip-172-16-101-9.us-west-2.compute.internal-auto-hep     ip-172-16-101-9.us-west-2.compute.internal     *           172.16.101.9,192.168.71.192      projectcalico-default-allow
+ip-172-16-102-63.us-west-2.compute.internal-auto-hep    ip-172-16-102-63.us-west-2.compute.internal    *           172.16.102.63,192.168.108.192    projectcalico-default-allow
+```
+
+#### Apply network policy to automatic host endpoints
+
+To apply policy that targets all Kubernetes nodes, first add a label to the nodes.
+The label will be synced to their automatic host endpoints.
+
+For example, to add the label **kubernetes-host** to all nodes and their host endpoints:
+
+```bash
+kubectl label nodes --all kubernetes-host=
+```
+
+And an example policy snippet:
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: all-nodes-policy
+spec:
+  selector: has(kubernetes-host)
+  <rest of the policy>
+```
+
+To select a specific set of host endpoints (and their corresponding Kubernetes nodes), use a policy selector that selects a label unique to that set of host endpoints.
+For example, if we want to add the label **environment=dev** to nodes named node1 and node2:
+
+```bash
+kubectl label node node1 environment=dev
+kubectl label node node2 environment=dev
+```
+
+With the labels in place and automatic host endpoints enabled, host endpoints for node1 and node2 will be updated with the **environment=dev** label.
+We can write policy to select that set of nodes with a combination of selectors:
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: some-nodes-policy
+spec:
+  selector: has(kubernetes-host) && environment == 'dev'
+  <rest of the policy>
+```
+
+### Tutorial
+
+This tutorial will lock down Kubernetes node ingress to only allow SSH and required ports for Kubernetes to function.
+We will apply two policies: one for the master nodes. and one for the worker nodes.
+
+> Note: Run this tutorial only on a sandbox cluster; using it on a real cluster can disrupt traffic.
```

Comment:
This is not a good restriction.  We need to be supplying instructions that people can use in production clusters.

Reply:
I think you're right. And, we should assume admins will test out these changes in non-critical environments.

68860ae2940e1a3e317576b44a2eb34269e708e7